File: 240307/운행 되고 있는 시간/hours-in-service.py
n=int(input())

work_t=[list(map(int,input().split())) for _ in range(n)]

time=[0]*10

for t in work_t:
    s,e=t[0],t[1]
    for i in range(s,e):
        time[i]+=1
working=0

for i in time:
    if i!=0:
        working+=1

def get_time(i):
    cur=working
    for j in range(work_t[i][0],work_t[i][1]):
        if time[j]-1==0:
            cur-=1
    return cur 

ans=0
for i in range(n):
    t=get_time(i) 
    ans=max(ans,t)

print(ans)

File: 260118/운행 되고 있는 시간/hours-in-service.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");

const n = Number(input[0]);
const segments = input.slice(1, n + 1).map(line => line.split(' ').map(Number));

// Please Write your code here.
let ans = 0;

for(let i=0; i<n; i++){
    let arr = Array(1001).fill(0);
    for(let j=0; j<n; j++){
        if(i===j) continue;
        const [t1, t2] = segments[j];
        for(let k=t1; k<t2; k++){
            arr[k]=1;
        }
        let cnt = 0;
        arr.forEach((n)=>{
            if(n===1) cnt++;
        })
        ans = Math.max(ans,cnt)
    }
}

console.log(ans);